File: week1-python/Basic_Image_Operations/Cropping_an_Image_section.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH

# Read image
image = cv2.imread(DATA_PATH+"images/boy.jpg")

# Let's see what image we are dealing with
#cv2.imshow("Image",image)
#cv2.waitKey(0)

# Crop out a rectangle
# x coordinates = 170 to 320
# y coordinates = 40 to 200
crop = image[40:200,170:320]
#cv2.imshow("Cropped Image",crop)
#cv2.waitKey(0)
cv2.imwrite("../results/croppedImage.png",crop)

File: week1-cpp/Basic_Image_Operations/Cropping_an_Image_Section.cpp
// Include libraries
#include <iostream>
#include "dataPath.hpp"
#include <opencv2/opencv.hpp>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(void){
	// Read image
	Mat image = imread(DATA_PATH+"images/boy.jpg");
	// Crop out a rectangle
	// x coordinates = 170 to 320
	// y coordinates = 40 to 200
	Mat crop = image(Range(40,200),Range(170,320));
	imwrite("../results/crop.png",crop);
	//imshow("Cropped image",crop);
	//waitKey(0);
	return 0;
}
